core/mobile-menu.ts

```diff
@@ -0,0 +1,47 @@
+import { atom, computed } from 'nanostores'
+
+import { fastCategories } from './fast.ts'
+import { router } from './router.ts'
+import { slowCategories } from './slow.ts'
+
+const isMenuShowed = atom<boolean>(false)
+
+const isMenuHasItems = computed(
+  [router, fastCategories, slowCategories],
+  (r, f, s) => {
+    if (r.route === 'fast') {
+      return !f.isLoading && f.categories.length > 1
+    }
+
+    if (r.route === 'slow') {
+      return !s.isLoading && s.tree.length > 1
+    }
+
+    return true
+  }
+)
+
+export function toggleMenu(): void {
+  isMenuShowed.set(!isMenuShowed.get())
+}
+
+export const isMenuOpened = computed(
+  [isMenuShowed, isMenuHasItems],
+  (isShowed, isHasItems) => {
+    return isShowed && isHasItems
+  }
+)
+
+router.subscribe((newRoute, prevRoute) => {
+  let targetRoutes = ['add', 'slow', 'fast']
```

Comment:
Let’s do a micro-optimization here by moving it to the root (we don’t need to define new set every time) and use object or `Set`